parsing/unit_info.ml

```diff
@@ -32,36 +35,38 @@ let basename_chop_extensions basename  =
   | dot_pos -> String.sub basename 0 dot_pos
   | exception Not_found -> basename
 
-let modulize s = String.capitalize_ascii s
+let strict_modulize s =
+  match Misc.UString.capitalize s with
+  | Ok x -> x
+  | Error _ -> raise (Error (Invalid_encoding s))
 
-(* We re-export the [Misc] definition *)
-let normalize = Misc.normalized_unit_filename
+let modulize s = match Misc.UString.capitalize s with Ok x | Error x -> x
 
-let modname_from_source source_file =
-  source_file |> Filename.basename |> basename_chop_extensions |> modulize
+(* We re-export the [Misc] definition, and ignore encoding errors under the
+   assumption that we should focus our effort on not *producing* badly encoded
+   module names *)
+let normalize x = match Misc.normalized_unit_filename x with |
```

Comment:
The docstring should be updated to say that it also apply NFC normalization to some characters.

Reply:
I have updated the documentation of the `Misc.normalized_unit_filename` function, which seemed to be a more natural place for mentioning the NFC normalization.

Reply:
Looks good :)